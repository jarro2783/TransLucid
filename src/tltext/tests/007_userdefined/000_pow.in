op ** = OpInfix."exponent".false.AssocRight.400;;
op ^  = OpInfix."exponent".false.AssocRight.400;;

fun exponent.a.b [a : intmp, b : intmp] = pow.b.a;;

fun pow.n = g
where
  dim d <- n;;
  var g = fby.d (\s -> 1) (\ {d} s -> s * g.s);;
end;;

%%
2 ^ 0;;
2 ^ 1;;
2 ^ 2;;
2 ^ 3;;
3 * 2 ^ 10;;
10 ^ (10 * 2);;
5 ** 6;;
5 ^ 6;;
