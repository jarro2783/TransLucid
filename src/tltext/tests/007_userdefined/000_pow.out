1
8
1
2
4
8
3072
100000000000000000000
15625
15625
